solver/src/driver.rs

```diff
@@ -197,29 +163,94 @@ impl Driver {
                 continue;
             }
 
-            let trades = settlement.trades().to_vec();
+            let trades = settlement.settlement.trades().to_vec();
             match settlement_submission::submit(
                 &self.settlement_contract,
                 self.gas_price_estimator.as_ref(),
                 self.target_confirm_time,
                 GAS_PRICE_CAP,
-                settlement,
+                settlement.settlement,
             )
             .await
             {
                 Ok(_) => {
-                    // TODO: order validity checks
-                    // Decide what is handled by orderbook service and what by us.
-                    // We likely want to at least mark orders we know we have settled so that we don't
-                    // attempt to settle them again when they are still in the orderbook.
                     trades
                         .iter()
                         .for_each(|trade| self.metrics.order_settled(&trade.order));
                     break;
                 }
-                Err(err) => tracing::error!("{} Failed to submit settlement: {:?}", solver, err),
+                Err(err) => tracing::error!(
+                    "{} Failed to submit settlement: {:?}",
+                    self.solver[settlement.index],
+                    err,
+                ),
             }
         }
+    }
+
+    pub async fn single_run(&mut self) -> Result<()> {
+        tracing::debug!("starting single run");
+        let liquidity = self.liquidity_collector.get_liquidity().await?;
+
+        let estimated_prices = self.collect_estimated_prices(&liquidity).await;
+        let liquidity = liquidity_with_price(liquidity, &estimated_prices);
+        self.metrics.liquidity_fetched(&liquidity);
+
+        let gas_price = self
+            .gas_price_estimator
+            .estimate()
+            .await
+            .context("failed to estimate gas price")?;
+        tracing::debug!("solving with gas price of {}", gas_price);
+
+        let mut settlements = self
+            .run_solvers(liquidity, gas_price, &estimated_prices)
+            .await;
+        // Sort by key in descending order. Reversed by doing b.cmp(a) instead of a.cmp(b).
+        settlements.sort_unstable_by(|a, b| b.objective_value.cmp(&a.objective_value));
+
+        for settlement in settlements.iter() {
+            tracing::info!(
+                "solver {} found solution with objective value {}: {:?}",
+                self.solver[settlement.index],
+                settlement.objective_value,
+                settlement.settlement,
```

Comment:
This is now a single log entry that prints both the objective value and the settlement contents.